lib/tracecontext/index.js

```diff
@@ -0,0 +1,74 @@
+'use strict'
```

Comment:
This is our new `TraceContext` class, replacing the `TraceParent` class.  This object composes together a `TraceParent` and `TraceState` class to do its work (parsing of the header, starting a new trace if necessary, re-serializing the headers as strings for outgoing transactions)